instrumentation/opentelemetry-instrumentation-celery/src/opentelemetry/instrumentation/celery/__init__.py

```diff
@@ -108,7 +108,23 @@ def _trace_prerun(self, *args, **kwargs):
 
         logger.debug("prerun signal start task_id=%s", task_id)
 
-        span = self._tracer.start_span(task.name, kind=trace.SpanKind.CONSUMER)
+        (parent, publisher) = utils.retrieve_span_context(task)
+
+        links = []
+        if publisher is not None:
+            links.append(trace.Link(publisher))
+
+        if parent is not None:
+            span = self._tracer.start_span(
+                task.name,
+                parent=parent,
+                links=links,
+                kind=trace.SpanKind.CONSUMER,
+            )
```

Comment:
Can the else condition be removed and parent be passed to the function always. If it is None, it shouldn't have any effect on the span right? Rest of the arguments seem to be exactly the same.